packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -527,6 +537,18 @@ export class EntityService {
 
     const entity = await this.acquireLock(args, user);
 
+    const newName = args.data.name?.toLowerCase().trim();
+    const newPluralDisplayName = args.data.pluralDisplayName
+      ?.toLowerCase()
+      .trim();
+    const currPluralDisplayName = entity?.pluralDisplayName;
+
+    if (newName === (newPluralDisplayName || currPluralDisplayName)) {
+      throw new AmplicationError(
+        `The entity name and plural display name cannot be the same.`
+      );
+    }
+
```

Comment:
the new just like the pluralDisplayName can have a new value or a current value
you probably better do something like this:

```
    const newName = args.data.name?.toLowerCase().?trim() || entity.name.toLowerCase().trim();
    const newPluralDisplayName = args.data.pluralDisplayName
      ?.toLowerCase()
      ?.trim() || entity.pluralDisplayName.toLowerCase().trim(); 

    if (newName === newPluralDisplayName) {
      throw new AmplicationError(
        `The entity name and plural display name cannot be the same.`
      );
    }
```